File: src/Day01.hs
module Main (main) where

-- import           Advent

main :: IO ()
main =
  do inp <- map read <$> lines <$> readFile "inputs/input01.txt"
     let pairs = [(x,y)|x<-inp,y<-[z|z<-inp,z<=2020-x],x+y==2020]
     let triples = [(x,y,w)|x<-inp,y<-[z|z<-inp,z<=2020-x],w<-[z|z<-inp,z<=2020-y-x],x+y+w==2020]
     print ((\(x,y) -> x*y) (head pairs))
     print ((\(x,y,z) -> x*y*z) (head triples))

File: src/Day01.py
file1 = open("inputs/input01.txt", "r")
lines = list(map(lambda x: int(x), file1.readlines()))
part1 = [x * y for x in lines for y in lines if (x + y) == 2020][0]
part2 = [x * y * z for x in lines for y in lines for z in lines if (x + y + z) == 2020][
    0
]
print(part1)
print(part2)
